test/e2e/customconfigs/reconcile_test.go

```diff
@@ -0,0 +1,147 @@
+			cfg.ManagedJobsNamespaceSelector = &v1.LabelSelector{
+				MatchExpressions: []v1.LabelSelectorRequirement{
+					{
+						Key:      "managed-by-kueue",
+						Operator: v1.LabelSelectorOpIn,
+						Values:   []string{"true"},
+					},
+				},
+			}
+		})
+
+		ns := &corev1.Namespace{}
+		err := k8sClient.Get(ctx, client.ObjectKey{Name: managedNamespace}, ns)
+		if apierrors.IsNotFound(err) {
+			ns = &corev1.Namespace{
+				ObjectMeta: v1.ObjectMeta{
+					Name: managedNamespace,
+					Labels: map[string]string{
+						"managed-by-kueue": "true",
+					},
+				},
+			}
+			err = k8sClient.Create(ctx, ns)
+		}
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		rf = testing.MakeResourceFlavor(rfName).Obj()
+		gomega.Expect(k8sClient.Create(ctx, rf)).To(gomega.Succeed())
+
+		cq = testing.MakeClusterQueue(cqName).
+			ResourceGroup(*testing.MakeFlavorQuotas(rfName).Resource(corev1.ResourceCPU, "5").Obj()).
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
+
+		lq = testing.MakeLocalQueue(lqName, managedNamespace).ClusterQueue(cqName).Obj()
+		gomega.Expect(k8sClient.Create(ctx, lq)).To(gomega.Succeed())
+	})
+
+	ginkgo.AfterAll(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, cq, true, util.LongTimeout)
+		util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, rf, true, util.LongTimeout)
+	})
+
+	ginkgo.It("should not reconcile a job in the default (unmanaged) namespace", func() {
+		job := testingjob.MakeJob("unmanaged-job", v1.NamespaceDefault).
+			Queue(kueue.LocalQueueName(lq.Name)).
+			Suspend(true).
+			Image(util.GetAgnHostImage(), util.BehaviorExitFast).
+			Obj()
+
+		gomega.Expect(k8sClient.Create(ctx, job)).To(gomega.Succeed())
+
+		wls := &kueue.WorkloadList{}
+		gomega.Expect(k8sClient.List(ctx, wls, client.InNamespace(v1.NamespaceDefault))).To(gomega.Succeed())
+		gomega.Expect(wls.Items).To(gomega.BeEmpty(), "Expected no workload in unmanaged namespace")
+	})
+
+	ginkgo.It("should reconcile a job in managed namespace and create a workload", func() {
+		job := testingjob.MakeJob("managed-job", managedNamespace).
+			Queue(kueue.LocalQueueName(lq.Name)).
+			Suspend(true).
+			Image(util.GetAgnHostImage(), util.BehaviorExitFast).
+			Obj()
+
+		gomega.Expect(k8sClient.Create(ctx, job)).To(gomega.Succeed())
+
+		ginkgo.By("check that only one workload is created and admitted", func() {
+			createdWorkloads := &kueue.WorkloadList{}
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.List(ctx, createdWorkloads, client.InNamespace(managedNamespace))).To(gomega.Succeed())
+				g.Expect(createdWorkloads.Items).To(gomega.HaveLen(1))
+				g.Expect(createdWorkloads.Items[0].Status.Conditions).To(testing.HaveConditionStatusTrue(kueue.WorkloadAdmitted))
+			}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
```

Comment:
why do we need longtimeout?
could we use just a timeout?

Reply:
TBH, we probably could use timeout, and 99% is below 5s, but sometimes the testing machines are under load, and we had to bump it in a couple of places.

Actually, I think it is fair, to just check if workload is created, and move on. Then 10s (Timeout) seems safe to me.